File: src/network/packet/CloudboundPacket.php
<?php

namespace pocketcloud\cloud\network\packet;

use pocketcloud\cloud\network\packet\util\PacketData;

/**
 *  CloudboundPacket -> Cloud is the receiver (only decode), Server (Client) is the sender
 */
interface CloudboundPacket extends Packet {

    public function decodePayload(PacketData $packetData): void;
}

File: src/main/java/de/pocketcloud/cloud/network/packet/CloudboundPacket.java
package de.pocketcloud.cloud.network.packet;

import de.pocketcloud.cloud.network.packet.data.PacketData;

/**
 * Marker interface for packets that are sent from the Client (Server) to the Cloud.
 * CloudboundPacket -> Cloud is the receiver, Server (Client) is the sender
 */
public interface CloudboundPacket extends Packet {

    @Override
    void encodePayload(PacketData packetData);

    @Override
    default void decodePayload(PacketData packetData) {}
}